app/models/miq_provision_request_template.rb

```diff
@@ -48,4 +50,29 @@ def owner_options(service_task)
       :owner_last_name  => user.last_name
     }
   end
+
+  def number_of_vms(service_task, parent_svc, template_service_resource)
+    parent_task = MiqRequestTask.find_by_id(service_task.options[:parent_task_id])
+    resource = template_service_resource.resource
```

Comment:
The variable `resource` does not get referenced until you are inside the `if template_service_resource` block.  This assignment can be moved within the block.